### DISCUSSION TITLE
HW 2 Part 2 2B

### DISCUSSION THREAD

**Discussion by FishMacaron** [cooperrc/computational-mechanics#43] [Q&A] [answered]
This is what I currently have for 2B, I'm a bit lost about how to code this part. I'm pretty sure I will have to find all the *MOS transistor count* data per *Design* and then plot the *Date of Introduction* on the x axis while having *MOS transistor count* on y.
**My main issue is finding a way to section out the *MOS transistor count* data per category of *Design*.** And then using plt.semilogy correctly but that is not my biggest concern as of right now.

I'd assume one way could be to use something similar to DOI17=data[data['Date of Introduction']==2017] (this is my command when I was extracting all the data for 2017) but I wouldn't want to manually do this for every type of design. I feel like there would be a more efficient way to do it because it would be a nightmare to attempt graphing all those variables manually.

I'm trying to reference the *Beer ABV vs. IBU mean values by style* graph from the Visualizing Multiple Data > Bubble chart section to make it work but I'm not really sure where to go from where I am. 

This is where I am right now:
![image](https://user-images.githubusercontent.com/68212585/172754407-3596bf37-cfc2-4e1e-b90f-d826e3388016.png)

**Comment by cooperrc** [comment, marked as answer]
Here, you don't need to count the styles/manufacturers. You need to plot one line per manufacturer (transistor count vs date). 

For example, if I wanted to plot IBM and Intel I could make 2 pandas dataframes:

```python
IBM = data[data['Designer'] == "IBM"]
Intel = data[data['Designer'] == "Intel"]
plt.semilogy(IBM['Date of Introduction'], IBM['MOS transistor count'], label = "IBM")
plt.semilogy(Intel['Date of Introduction'], Intel['MOS transistor count'], label = "Intel")
plt.xlabel('Date of introduction')
plt.ylabel('transistor count')
plt.legend();
```

If you want an array of all of the designers, use `data['Designer'].unique()`
